controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -119,6 +121,33 @@ func (w *Workload) getConfigMap(ctx context.Context, configMap ctrlclient.Object
 	return original.DeepCopy(), nil
 }
 
+var managementClusterVersionCeiling = semver.MustParse("1.22.0")
+
+// IsKubernetesVersionSupported checks if the Kubernetes version is supported.
+// Kubernetes versions >= v1.22.0 for management clusters are not supported.
+// Management clusters are identified by the existence of the KubeadmControlPlane CRD.
+func (w *Workload) IsKubernetesVersionSupported(ctx context.Context, version semver.Version) (bool, error) {
+	// Kubernetes version is < v1.22.0.
+	if version.LT(managementClusterVersionCeiling) {
+		return true, nil
+	}
+
+	// Try to get *metav1.PartialObjectMetadata of KubeadmControlPlane CRD.
+	out := &metav1.PartialObjectMetadata{}
+	out.SetGroupVersionKind(apiextensionsv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"))
```

Comment:
Not sure if it's worth putting that part into a top-level var

Reply:
Looks good for me!